tests fail

11 tests fail for the latest release using Python 2.7 from Debian/unstable.

Unless fixed, this means nose2 and everything that depends on nose2 will get removed from Debian.

Example:

```
======================================================================
FAIL: test_failure_to_read_empty_properties (nose2.tests.functional.test_junitxml_plugin.JunitXmlPluginFunctionalFailureTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "nose2/tests/functional/test_junitxml_plugin.py", line 167, in test_failure_to_read_empty_properties
    "'.*%s'" % filename_for_regex)
  File "nose2/tests/_common.py", line 64, in assertTestRunOutputMatches
    testf(util.safe_decode(cmd_stderr), stderr)
AssertionError: Regexp didn't match: "Internal Error: runTests aborted: JUnitXML: could not decode file: '.*empty_properties/properties.json'" not found in u"test (test_junitxml_empty_properties.Test) ... ok\nInternal Error: runTests aborted: [Errno 2] JUnitXML: Properties file does not exist: '/home/brian/tree/debian/python-modules/nose2/nose2/tests/functional/support/scenario/junitxml/empty_properties/properties.json'\n"

```

For the tests you need to install `mock`. It is not listed in the requirements.txt because it's needed only for the tests.

After installing mock, I'm down to one failing test, (`nose2.tests.functional.test_main.TestPluggableTestProgram.test_run_in_module_from_its_main`) with this stacktrace:

```
Traceback (most recent call last):
  File "nose2/tests/functional/test_main.py", line 27, in test_run_in_module_from_its_main
    self.assertTestRunOutputMatches(proc, stderr='Ran 1 test')
  File "nose2/tests/_common.py", line 64, in assertTestRunOutputMatches
    testf(util.safe_decode(cmd_stderr), stderr)
AssertionError: Regexp didn't match: 'Ran 1 test' not found in u'Traceback (most recent call last):\n  File "/home/corentih/tmp/nose2/nose2/tests/functional/support/scenario/one_test/tests.py", line 1, in <module>\n    import nose2\nImportError: No module named nose2\n'
```

Weirdly, this test does `import nose2`, so you need to install it. After installing it, I have no more failure.

Still one test failure - my fault???

```
======================================================================
FAIL: test_run (nose2.tests.functional.test_coverage.TestCoverage)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "nose2/tests/functional/test_coverage.py", line 24, in test_run
    stderr=os.path.join('lib', 'mod1.py').replace('\\', r'\\') + STATS)
  File "nose2/tests/_common.py", line 64, in assertTestRunOutputMatches
    testf(util.safe_decode(cmd_stderr), stderr)
AssertionError: Regexp didn't match: 'lib/mod1.py           8      5    38%' not found in u'test1 (test_coverage.TestLib) ... ok\n\n----------------------------------------------------------------------\nRan 1 test in 0.001s\n\nOK\n---------- coverage: platform linux2, python 2.7.11-final-0 ----------\nName           Stmts   Miss  Cover\n----------------------------------\nlib/__init__       0      0   100%\nlib/mod1           8      5    38%\n----------------------------------\nTOTAL              8      5    38%\n'

```

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -1,6 +1,12 @@
 Changelog
 =========
 
+0.6.2
+-----
+
+* Fixed
+    * fix the coverage plugin tests for coverage==3.7.1
+
 0.6.1
 -----
 

--- a/nose2/tests/functional/test_coverage.py
+++ b/nose2/tests/functional/test_coverage.py
@@ -16,12 +16,14 @@ class TestCoverage(FunctionalTestCase):
             '--with-coverage',
             '--coverage=lib/'
         )
-        STATS = '           8      5    38%'
-
+        STATS = '\s+8\s+5\s+38%'
+        expected = os.path.join('lib', 'mod1(.py)?')
+        expected = expected.replace('\\', r'\\')
+        expected = expected + STATS
         stdout, stderr = proc.communicate()
         self.assertTestRunOutputMatches(
             proc,
-            stderr=os.path.join('lib', 'mod1.py').replace('\\', r'\\') + STATS)
+            stderr=expected)
         self.assertTestRunOutputMatches(
             proc,
-            stderr='TOTAL      ' + STATS)
+            stderr='TOTAL\s+' + STATS)

--- a/setup.py
+++ b/setup.py
@@ -2,7 +2,7 @@ import os
 import sys
 
 NAME = 'nose2'
-VERSION = '0.6.1'
+VERSION = '0.6.2'
 PACKAGES = ['nose2', 'nose2.plugins', 'nose2.plugins.loader',
             'nose2.tests', 'nose2.tests.functional', 'nose2.tests.unit',
             'nose2.tools', 'nose2.backports']

